assets/src/edit-story/components/library/panes/media/mediaUtils.js

```diff
@@ -85,6 +85,140 @@ export const getResourceFromMediaPicker = (mediaPickerEl) => {
   };
 };
 
+/**
+ * Generates a virtual image to pick metadata resource
+ *
+ * @param {Object} image File object
+ * @return {Promise} Resource object
+ */
+
+const getImageMetadata = (image) => {
+  const reader = new window.FileReader();
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function(e) {
+      const src = e.target.result;
+      const randomId = Math.floor(Date.now() / 1000);
+
+      const img = new window.Image();
+
+      img.onload = function() {
+        const width = img.width;
+        const height = img.height;
+
+        resolve({
+          type: 'image',
+          src,
+          width,
+          height,
+          mimeType: image.type,
+          posterId: randomId,
+          poster: '',
+          videoId: undefined,
+          local: true,
+          lengthFormatted: undefined,
+        });
+      };
+
+      img.src = src;
+    };
+
+    reader.onerror = reject;
+
+    reader.readAsDataURL(image);
+  });
+};
+
+/**
+ * Generates a virtual video to pick metadata resource
+ *
+ * @param {Object} video File object
+ * @return {Promise} Resource object
+ */
+
+const getVideoMetadata = (video) => {
+  const reader = new window.FileReader();
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function() {
+      const blob = new window.Blob([reader.result], { type: video.type });
+      const src = URL.createObjectURL(blob);
+      const v = document.createElement('video');
+      const randomId = Math.floor(Date.now() / 1000);
+
+      const snapshot = () => {
+        const canvas = document.createElement('canvas');
```

Comment:
This code already exists. We should reuse.

Reply:
Awesome! Thanks for clarify that. 👍

Reply:
Same as https://github.com/google/web-stories-wp/blob/53c46223274f5a920c506c19f611ee52d038f162/assets/src/edit-story/app/media/utils/getFirstFrameOfVideo.js

Reply:
Yeah! I see that code, thanks for share @spacedmonkey